cli/azd/pkg/project/framework_service_external.go

```diff
@@ -464,8 +464,14 @@ func (efs *ExternalFrameworkService) toProtoServiceConfig(serviceConfig *Service
 		return nil, nil
 	}
 
+	// Use an empty resolver since ExternalFrameworkService doesn't have access to environment
+	// The extension is responsible for handling environment variable substitution
+	emptyResolver := func(key string) string {
+		return ""
+	}
```

Comment:
If we need an environment we can inject it into the framework service otherwise this is fine.